src/error.rs

```diff
@@ -0,0 +1,33 @@
+use core::fmt::{Display, Formatter};
+use fmt;
+
+/// Describes the error that can happen when the
+/// "multiboot2 information structure" (*mbi*) gets loaded.
+#[derive(Debug)]
+pub enum MbiLoadError {
+    /// The address is invalid. Make sure that the address is 8-byte aligned,
+    /// according to the spec.
+    IllegalAddress,
+    /// The total size of the multiboot2 information structure must be a multiple of 8.
+    /// (Not in spec, but it is implicitly the case, because the begin of MBI
+    /// and all tags are 8-byte aligned and the end tag is exactly 8 byte long).
+    IllegalTotalSize(u32),
+    /// End tag missing. Each multiboot2 header requires to have an end tag.
+    NoEndTag,
+}
+
+impl Display for MbiLoadError {
```

Comment:
Hmm, I'm not sure implementing `Display` for errors like this adds much over the `Debug` implementation

Reply:
I don't mind if we don't have it. BTW: My general rule is that if in doubt to implement a little more than a little to less. There is no disadvantage in having it. @IsaacWoods